matchbox_socket/src/webrtc_socket/mod.rs

```diff
@@ -160,26 +161,29 @@ async fn message_loop<M: Messenger>(
 
         select! {
             _  = &mut timeout => {
-                requests_sender.unbounded_send(PeerRequest::KeepAlive).expect("send failed");
-                // UNWRAP: we will only ever get here if there already was a timeout
-                let interval = keep_alive_interval.unwrap();
-                timeout = Either::Left(Delay::new(interval)).fuse();
+                requests_sender.unbounded_send(PeerRequest::KeepAlive).map_err(TrySendError::into_send_error)?;
+                if let Some(interval) = keep_alive_interval {
+                    timeout = Either::Left(Delay::new(interval)).fuse();
+                }
             }
 
             message = events_receiver.next().fuse() => {
                 if let Some(event) = message {
                     debug!("{event:?}");
                     match event {
                         PeerEvent::IdAssigned(peer_uuid) => {
-                            id_tx.try_send(peer_uuid.to_owned()).unwrap();
+                            id_tx.take().expect("already sent peer id").send(peer_uuid.to_owned()).map_err(MessageSendError::PeerId)?;
```

Comment:
Does this need to be an error? As far as I understand, it would only happen if you drop the socket struct and continue polling the loop. Perhaps better to simply `warn`?

Reply:
So, I removed crossbeam channels and uses a oneshot channel to deliver PeerId. When it first gets the PeerId from the client, it takes the channel and uses it to send the peer Id. 

I actually think the only time this error gets triggered is when the receiver closed (dropped). I put it as an error here but does it make more sense as a panic?

Reply:
I think perhaps it just makes sense to stop the loop, without any error or panic. It goes back to what we intend to be the supported way of stopping a socket.

Reply:
Fixed